personalrobotics/ada_feeding
# Implement Full Schema using MoveIt2Servo

# Description

Implement the full acquisition schema.

# Testing procedure

Tested w/ Action 0 (i.e. just straight down and up).

* Pull the updated `pr_ros_controllers`: https://github.com/personalrobotics/pr_ros_controllers/pull/30
* Tested with dummy carrots in sim
* Tested with real carrots.

Observed occasional errors:
* Misses the carrot (camera calibration? or detection before robot has finished moving above plate, maybe add an 0.5s delay?)
* Delay on twist execution means that the robot does not move as far as expected during extraction. Okay for now, need to investigate later
* Carrot depth occasionally off (TODO later: verify against depth camera)

# Before opening a pull request

# Before Merging
